### DISCUSSION TITLE
MaRCoS Python packaging

### DISCUSSION THREAD

**Discussion by M4a1x** [vnegnev/marcos_extras#10] [Ideas]
Hi there!
I only went through the first tutorials (setting up the simulator, the client with the simulator, writing first sequence and running it on the marga simulator and testing the connection to the server on the RedPitaya) and had a few issues with the Python setup/importing/paths.
To mitigate those problems for others I played around with my fork of the marcos_client repo over the weekend to try to make it into a "normal" importable Python package as mentioned here; https://github.com/vnegnev/marcos_client/issues/12 This does include some "breaking" changes (e.g. path structure, how to use it) that are not documented yet and since I've only done the basic tutorials I only verified that those are still working (even though there are quite a few other files...). I'd be happy to hear some feedback before going much deeper on this!
Currently working (from a test remote) is a: `pip install marcos`, in the file: `from marcos import Experiment` and then using the `Experiment` class like before, e.g. `experiment = Experiment(lo_freq=5, rx_t=3.125)` for the marga simulator.

**Comment by vnegnev** [comment]
I think this is a really useful change - I'll try to have a play in the next few weeks, but I'm very short on time at the moment so please don't wait for me! My knowledge of Python packaging is quite basic, which is the main reason `marcos_client` uses such crude path management. Also @catkira originally suggested making a package - he might be interested in testing your package out.
